src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ObjectCreationExpression.cs

```diff
@@ -183,7 +183,8 @@ public override BoundNode VisitWithExpression(BoundWithExpression withExpr)
                     Debug.Assert(left.MemberSymbol is not null);
 
                     // We evaluate the values provided in source first
-                    BoundLocal valueTemp = _factory.StoreToTemp(assignment.Right, out BoundAssignmentOperator boundAssignmentToTemp);
+                    var rewrittenRight = (BoundExpression)Visit(assignment.Right)!;
```

Comment:
Consider using `LocalRewriter.VisitExpression` which should make the cast and the `!` unnecessary